### PR TITLE
refactor: refactor some code

### PR DESCRIPTION
**Pull request by hiroppy**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [x] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Improve readability.

### Breaking Changes

no

### CODE REVIEW

File: src/utils/setupWriteToDisk.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
 import fs from 'fs';
 import path from 'path';
 
+import { pluginName } from '../constants';
```
Comment (evilebottnawi) [inline src/utils/setupWriteToDisk.js:4]: No need extra require for one string, keep it as is in source code

File: src/utils/setupHooks.js [hunk 2]
```diff
@@ -73,7 +75,7 @@ export default function setupHooks(context) {
     });
   }
 
-  context.compiler.hooks.watchRun.tap('DevMiddleware', invalid);
-  context.compiler.hooks.invalid.tap('DevMiddleware', invalid);
-  context.compiler.hooks.done.tap('DevMiddleware', done);
+  context.compiler.hooks.watchRun.tap(pluginName, invalid);
+  context.compiler.hooks.invalid.tap(pluginName, invalid);
+  context.compiler.hooks.done.tap(pluginName, done);
```
Comment (evilebottnawi) [inline src/utils/setupHooks.js:80]: No need extra require, we don't change these lines very often

File: src/utils/getFilenameFromUrl.js [hunk 3]
```diff
@@ -23,11 +23,10 @@ function getPaths(stats, options) {
   return publicPaths;
 }
 
-const memoizedParse = mem(parse);
-
-export default (context, url) => {
+export default function getFilenameFromUrl(context, url) {
   const { options } = context;
   const paths = getPaths(context.stats, options);
+  const memoizedParse = mem(parse);
```
Comment (evilebottnawi) [inline src/utils/getFilenameFromUrl.js:29]: Memorize is broken, no need create it every time on each call

File: src/utils/getFilenameFromUrl.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import querystring from 'querystring';
 import mem from 'mem';
 
 function getPaths(stats, options) {
-  const childStats = stats.stats ? stats.stats : [stats];
+  const childStats = Array.isArray(stats.stats) ? stats.stats : [stats];
```
Comment (evilebottnawi) [inline src/utils/getFilenameFromUrl.js:8]: Extra check, `stats.stats` exists only for mutli compiler mode

File: src/middleware.js [hunk 5]
```diff
@@ -31,55 +39,42 @@ export default function wrapper(context) {
...
-
-        res.send(content);
-
-        resolve();
       }
 
-      ready(context, processRequest, req);
-    });
+      res.send(content);
+    }
```
Comment (evilebottnawi) [inline src/middleware.js:78]: This is the most important part and any change here is critical, very difficult to read

File: src/middleware.js [hunk 6]
```diff
@@ -31,55 +39,42 @@ export default function wrapper(context) {
...
-        const { headers } = context.options;
-
-        if (headers) {
-          for (const name in headers) {
-            if ({}.hasOwnProperty.call(headers, name)) {
-              res.set(name, context.options.headers[name]);
-            }
+      if (headers) {
+        for (const name in headers) {
+          if (Reflect.has(headers, name)) {
```
Comment (evilebottnawi) [inline src/middleware.js:71]: Bad idea

Comment (hiroppy) [reply src/middleware.js:71]: Why?

File: src/middleware.js (hunk 5 above)
Comment (evilebottnawi) [reply src/middleware.js:78]: Maybe you ca add comments before main steps?

Comment (hiroppy) [reply src/middleware.js:78]: Could you give me a comment? So you mean move `res.send(content);` to prev position?

File: src/middleware.js (hunk 6 above)
Comment (hiroppy) [reply src/middleware.js:71]: `{}.hasOwnProperty.call(headers, name)` is very poorly readable.

File: src/middleware.js (hunk 5 above)
Comment (evilebottnawi) [reply src/middleware.js:78]: No, just add comments something like:
```
// Check headers
// Search file in compilation
// Setup content-length for file
// Sending file
```

File: src/middleware.js (hunk 6 above)
Comment (evilebottnawi) [reply src/middleware.js:71]: Hm, looks `Reflect` has good supporting, so it is good to use it

Comment (hiroppy) [reply src/middleware.js]: Actually, this code is unnecessary. I updated.

File: src/middleware.js (hunk 5 above)
Comment (hiroppy) [reply src/middleware.js:79]: If you have points which are hard to read for you, please tell me.

**Comment by hiroppy** [comment]
azure-pipeline seems not good. It can't download Node.js image.

https://dev.azure.com/webpack/webpack-dev-middleware/_build/results?buildId=9126&view=logs&j=1d505e51-909d-5717-0d5b-43f6a0bd49f6&t=d7ba6858-798d-59e4-3be2-6e44963833ea

**Comment by evilebottnawi** [comment]
@hiroppy Let's resolve the PR with migration on github actions

**Comment by hiroppy** [comment]
@evilebottnawi This pr has already been done. Please review.
https://github.com/webpack/webpack-dev-middleware/pull/611

**Comment by hiroppy** [comment]
CI is green.

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**